### PR TITLE
Bug 1873900: pkg/cvo/sync_worker: Grow logging in SyncWorker.Update

### PR DESCRIPTION
**Pull request by wking**
To make it easier to debug situations where the CVO's `Operator.sync` logs:

    Desired version from spec is ...

but the sync worker itself doesn't seem to pick up the update (like [this][1] and [this][2]).

[1]: https://bugzilla.redhat.com/show_bug.cgi?id=1843505#c19
[2]: https://bugzilla.redhat.com/show_bug.cgi?id=1873900

### CODE REVIEW

**Comment by wking** [comment]
[e2e-upgrade][1] died on install.

/retest

[1]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cluster-version-operator/451/pull-ci-openshift-cluster-version-operator-master-e2e-upgrade/1302006145614876672

**Comment by wking** [comment]
[e2e-upgrade][1] was slow/disruptive, but did actually happen.  So trying again to see if we can get an ignored update...

/retest

[1]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cluster-version-operator/451/pull-ci-openshift-cluster-version-operator-master-e2e-upgrade/1303472915366285312

File: pkg/cvo/sync_worker.go [hunk 1]
```diff
@@ -213,7 +213,13 @@ func (w *SyncWorker) Update(generation int64, desired configv1.Update, overrides
 		w.work.Generation = generation
 	}
 
-	if work.Empty() || equalSyncWork(w.work, work) {
+	if work.Empty() {
+		klog.V(5).Info("Update work is empty; ignoring requested change")
```
Comment (LalatenduMohanty) [inline pkg/cvo/sync_worker.go:217]: `Update work is empty` is confusing to me. Can you expand this a little?

**Review by LalatenduMohanty — CHANGES_REQUESTED** [review]

File: pkg/cvo/sync_worker.go (hunk 1 above)
Comment (LalatenduMohanty) [reply pkg/cvo/sync_worker.go:217]: What about `Update work is empty i.e. there is no version to update`

Comment (wking) [reply pkg/cvo/sync_worker.go]: Updated to `Update work has no release image; ignoring requested change` with 0802f07fb0 -> 48b59d1414.

**Review by LalatenduMohanty — APPROVED** [review]
/lgtm

**Pull request merged**